replace unmaintained satori/go.uuid with google/uuid and bump x/net

tuminoid opened the pull request:
Bump go dependencies to get rid of public vulnerabilities, and to replace unmaintained UUID package with maintained one.

Following dependencies are bumped, the rest of the change is from tidy.

x/net 0.0.0-20220722155237-a158d28d115b to 0.8.0:
- https://osv.dev/vulnerability/GHSA-69cg-p879-7622
- https://osv.dev/vulnerability/GHSA-fxg5-wq6x-vr4w
- https://osv.dev/vulnerability/GO-2022-1144
- https://osv.dev/vulnerability/GHSA-vvpx-j8f3-3w6h

satori/go.uuid is unmaintained and is replaced commonly by `google/uuid`, requiring minor changes to the Go code as well:
- https://osv.dev/vulnerability/GO-2020-0018 

aws-sdk-go does not have a patch available.

tuminoid commented:
/cc @BenTheElder

BenTheElder commented:
/ok-to-test

BenTheElder commented on prow/pjutil/pjutil.go:
```diff
@@ -42,14 +42,15 @@ func NewProwJob(spec prowapi.ProwJobSpec, extraLabels, extraAnnotations map[stri
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        uuid.NewV1().String(),
+			Name:        uuidV1.String(),
```
/assign @cjwagner

tuminoid replied:
Changed to using `google/uuid` as suggested, and diff is even smaller `uuid.NewV1().String()` => `uuid.New().String()`

tuminoid replied:
And reworked to be V1, and not V4, so diff is back.

aojea commented:
kubernetes/kubernetes uses https://github.com/google/uuid , I think that is better to be consistent

tuminoid commented:
> kubernetes/kubernetes uses https://github.com/google/uuid , I think that is better to be consistent

`google/uuid` has even more compatible API, so the diff is even simpler. Thanks for suggestion, fixed.

aojea commented on prow/pjutil/pjutil.go:
```diff
@@ -49,7 +49,7 @@ func NewProwJob(spec prowapi.ProwJobSpec, extraLabels, extraAnnotations map[stri
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        uuid.NewV1().String(),
+			Name:        uuid.New().String(),
```
to check with the owners if is ok to change the version, this New() seems to generate a version 4.

For version 1 the generator seems to be NewUUID() https://github.com/google/uuid/blob/44b5fee7c49cf3bcdf723f106b36d56ef13ccc88/version1.go#L19

tuminoid replied:
Good catch, I thought I checked it did V1, but probably lost my thought somewhere along the way. Fixed.

aojea commented on kubetest/aksengine_helpers.go:
```diff
@@ -347,7 +347,7 @@ func (az *AzureClient) AssignOwnerRoleToIdentity(ctx context.Context, resourceGr
 	}
 
-	if _, err := az.authorizationClient.Create(ctx, scope, uuid.NewV1().String(), roleAssignmentParameters); err != nil {
+	if _, err := az.authorizationClient.Create(ctx, scope, uuid.New().String(), roleAssignmentParameters); err != nil {
```
https://github.com/kubernetes/test-infra/pull/29344/files#r1174906896

tuminoid replied:
Fixed.

tuminoid commented on prow/test/integration/test/horologium_test.go:
```diff
@@ -160,7 +160,7 @@ func TestLaunchProwJob(t *testing.T) {
 			pjToBe := pj.DeepCopy()
 			pjToBe.ResourceVersion = ""
-			pjToBe.Name = uuid.NewV1().String()
+			pjToBe.Name = uuid.New().String()
```
Fixed

tuminoid commented:
> > kubernetes/kubernetes uses https://github.com/google/uuid , I think that is better to be consistent
> 
> `google/uuid` has even more compatible API, so the diff is even simpler. Thanks for suggestion, fixed.

Reworked to use `NewUUID()` as suggested by @aojea to keep it V1. Diff is now the same as with `gofrs` version initially, but using `google/uuid`.

aojea commented:
lgtm for the dependencies mechanics, but someone more familiar should double check the uuid changes

tuminoid commented:
> lgtm for the dependencies mechanics, but someone more familiar should double check the uuid changes

Any idea who that someone might be?

tuminoid commented:
/cc @BenTheElder @aojea @chaodaiG @stevekuznetsov @spiffxp 

LGTM anyone?

aojea commented:
/lgtm

stevekuznetsov commented on prow/pjutil/pjutil.go:
```diff
@@ -42,14 +42,15 @@ func NewProwJob(spec prowapi.ProwJobSpec, extraLabels, extraAnnotations map[stri
 	labels, annotations := decorate.LabelsAndAnnotationsForSpec(spec, extraLabels, extraAnnotations)
 	specCopy := spec.DeepCopy()
 	setReportDefault(specCopy)
+	uuidV1, _ := uuid.NewUUID()
```
Documentation reads:

> NewUUID returns a Version 1 UUID based on the current NodeID and clock sequence, and the current time. If the NodeID has not been set by SetNodeID or SetNodeInterface then it will be set automatically. If the NodeID cannot be set NewUUID returns nil. If clock sequence has not been set by SetClockSequence then it will be set automatically. If GetTime fails to return the current NewUUID returns nil and an error.
> In most cases, New should be used. 

However, `New()` will panic. Let's continue using `NewUUID()` but check for `nil` and an error.

tuminoid replied:
Thanks for review! This `pjutil.go` function `NewProwJob` is returning `prowapi.ProwJob`, and won't accept returning error message, err, or nil. I switched it to `New()`.

Same with the `horologium_test.go`, I think we can panic in tests if UUID generation fails.

`aksengine_helpers.go` is now returning error string.  

Reading off `satori/go.uuid` it really didn't handle errors anywhere, so I think if it had failed it would've panicked, so very little change in behavior I think?

BenTheElder commented:
Thanks!
